File: 2024/python/01/part1.py
v1 = []
v2 = []
with open("input.txt") as file:
    for line in file:
        numbers = line.strip().split()
        v1.append(int(numbers[0]))
        v2.append(int(numbers[-1]))
v1.sort()
v2.sort()
sum = 0
for num1, num2 in zip(v1, v2):
    sum += abs(num1 - num2)

print(sum)

File: 2024/perl/01/part1.pl
use strict;

open(FH, "input.txt") or die "couldn't open";
my @v1 = ();
my @v2 = ();
while (<FH>) {
    chomp;
    my ($a, $b) = split /\s+/;
    push(@v1,$a);
    push(@v2,$b);
}
my $sum = 0;

my @v1s = sort(@v1);
my @v2s = sort(@v2);
for (my $i = 0; $i < scalar(@v1s); $i++) {
    $sum += abs($v1s[$i] - $v2s[$i]);
}
print $sum."\n";




